networking/networking.go

```diff
@@ -229,23 +233,43 @@ func Load(podRoot string, podID *types.UUID) (*Networking, error) {
 	}, nil
 }
 
-func (n *Networking) GetDefaultIP() net.IP {
-	if len(n.nets) == 0 {
-		return nil
+// GetDefaultNet iterates through all loaded networks and returns either
+// the first network that has masquerading enabled, the last network in case
+// there is no masqueraded one, or an error if no network was loaded.
+func (n *Networking) GetDefaultNet() (*activeNet, error) {
+	numberNets := len(n.nets)
+	if numberNets == 0 {
+		return nil, fmt.Errorf("no networks found")
+	}
+	for i := numberNets - 1; i >= 0; i-- {
+		net := n.nets[i]
+		if net.IPMasq() {
+			return &net, nil
+		}
 	}
-	return n.nets[len(n.nets)-1].runtime.IP
+	return &n.nets[numberNets-1], nil
+}
+
+// GetDefaultIP uses GetDefaultNet() to determine the default network and then
+// returns the Pod's IP of that network.
+func (n *Networking) GetDefaultIP() (net.IP, error) {
+	net, err := n.GetDefaultNet()
+	if err != nil {
+		return nil, err
+	}
+	return net.runtime.IP, nil
 }
 
 func (n *Networking) GetDefaultHostIP() (net.IP, error) {
-	if len(n.nets) == 0 {
-		return nil, fmt.Errorf("no networks found")
+	net, err := n.GetDefaultNet()
+	if err != nil {
+		return nil, err
 	}
-	return n.nets[len(n.nets)-1].runtime.HostIP, nil
+	return net.runtime.HostIP, nil
 }
 
-// GetIfacesByIP searches for and returns the interfaces with the given IP
-// Disregards the subnet mask since not every net.IP object contains
-// On success it will return the list of found interfaces
+// GetDefaultIP uses GetDefaultNet() to determine the default network and then
+// returns the Host's IP of that network.
```

Comment:
It's even worse, thanks for catching though! Oh boy, never work on an empty stomach :)